go/libraries/doltcore/env/actions/branch.go

```diff
@@ -575,53 +605,55 @@ func overwriteRoot(ctx context.Context, head *doltdb.RootValue, tblHashes map[st
 	return head, nil
 }
 
-// checkWorkingSetCompatibility checks that the current working set is "compatible" with the dest working set.
+// checkoutWouldStompWorkingSetChanges checks that the current working set is "compatible" with the dest working set.
 // This means that if both working sets are present (ie there are changes on both source and dest branches),
 // we check if the changes are identical before allowing a clobbering checkout.
 // Working set errors are ignored by this function, because they are properly handled elsewhere.
-func checkWorkingSetCompatibility(ctx context.Context, dEnv *env.DoltEnv, branchRef ref.BranchRef, currentWs *doltdb.WorkingSet) error {
-	db := dEnv.DoltDB
-	destWsRef, err := ref.WorkingSetRefForHead(branchRef)
+func checkoutWouldStompWorkingSetChanges(ctx context.Context, dEnv *env.DoltEnv, branchRef ref.BranchRef) bool {
```

Comment:
Is swallowing errors here problematic? Or are we relying on subsequent code to break?
Might be safer to return `true` (as that will cause the calling code to report an error), but then we'll end up reporting the wrong error to the user.

Reply:
It kind of is, but if there's actually a persistent problem loading any of these entities we wouldn't even get this far.